hack/yamlfmt/yamlfmt.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// Package main is the entry point to format YAML files
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+func main() {
+	indent := flag.Int("indent", 2, "default indent")
+	flag.Parse()
+
+	if flag.NArg() > 0 {
+		for _, path := range flag.Args() {
+			sourceYaml, err := os.ReadFile(path)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "%s: %v\n", path, err)
+				continue
+			}
+			documents, err := fetchYaml(sourceYaml)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "%s: %v\n", path, err)
+				continue
+			}
+			writer, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0640)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "%s: %v\n", path, err)
+				continue
+			}
+
+			err = streamYaml(writer, indent, documents)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "%s: %v\n", path, err)
+				continue
+			}
+		}
+	}
+}
+
+func formatNode(node *yaml.Node, index int) {
+	if node.Kind == yaml.DocumentNode {
+		formatNode(node.Content[0], index)
+		return
+	}
+	if node.Kind == yaml.MappingNode {
+		for i := 0; i < len(node.Content); i += 2 {
+			formatNode(node.Content[i], i/2)
+			formatNode(node.Content[i+1], i/2)
+		}
+	}
+	if node.Kind == yaml.SequenceNode {
+		for i := 0; i < len(node.Content); i++ {
+			formatNode(node.Content[i], i)
+		}
+	}
+	// Add a newline before the comment if it's not already there
+	if index != 0 && node.HeadComment != "" && !strings.HasPrefix(node.HeadComment, "\n") {
+		node.HeadComment = "\n" + node.HeadComment
+	}
```

Comment:
```suggestion

```

It looks like the added blank lines contain indented spaces, which doesn't look good, so let's delete them.